tests/unit_tests/AuthenticationTest.php

```diff
@@ -177,4 +177,44 @@ public function testPublishableKeys()
         $this->assertEquals(200, $publishable_keys['httpStatus']);
         $this->assertEquals('pk_test_everythingIsUnderControl', $publishable_keys['httpResponse']['publishable_key']);
     }
+
+    /**
+     * Test the getClientIdAndSecretMask method.
+     *
+     * This test verifies that the getClientIdAndSecretMask method correctly retrieves
+     * the client ID and client secret mask from the user manager resource.
+     *
+     * @throws \Exception
+     */
+    public function testGetClientIdAndSecretMask()
+    {
+        $response = array(
+            array(
+                'client_id' => 'test_client_id',
+                'client_secret_mask' => 'test_secret_mask'
+            ),
+        );
+
+        $this->_requestMock
+            ->expects($this->once())
+            ->method('exec')
+            ->will($this->returnValue(json_encode($response)));
+
+        $this->_requestMock
+            ->expects($this->any())
+            ->method('getinfo')
+            ->will($this->returnCallback(function($option) {
+                switch($option) {
+                    case CURLINFO_HTTP_CODE:
+                        return 200;
+                }
+                return null;
+            }));
+
+        $result = Authentication::getClientIdAndSecretMask($this->_configuration);
+        $this->assertCount(1, $result);
+        $this->assertEquals('test_client_id', $result[0]['client_id']);
+        $this->assertEquals('test_secret_mask', $result[0]['client_secret_mask']);
+    }
+
```

Comment:
I think we also have to test when there is an error during the HTTP request.
Once again, I don't speak only for this method but for a big part of this lib, we don't have any test if something goes wrong with HTTP requests.

Reply:
this is depending on the coming improvement with the integration of mockery